keychain/btcwallet.go

```diff
@@ -392,13 +392,15 @@ func (b *BtcWalletKeyRing) ECDH(keyDesc KeyDescriptor,
 }
 
 // SignMessage signs the given message, double-SHA256 hashing it first, with the
-// private key described in the key descriptor.
+// private key described in the key locator.
 //
 // NOTE: This is part of the keychain.MessageSignerRing interface.
-func (b *BtcWalletKeyRing) SignMessage(keyDesc KeyDescriptor,
+func (b *BtcWalletKeyRing) SignMessage(keyLoc KeyLocator,
 	msg []byte) (*btcec.Signature, error) {
 
-	privKey, err := b.DerivePrivKey(keyDesc)
+	privKey, err := b.DerivePrivKey(KeyDescriptor{
+		KeyLocator: keyLoc,
```

Comment:
Doesn't this break an assumption in the pool client?

Reply:
Fortunately not, since on the RPC level we've always only supported the key locator and not key descriptor for the `SignMessage{Compact}` calls.